drivers/gpio/gpio_stm32.c

```diff
@@ -194,37 +195,41 @@ static int gpio_stm32_init(struct device *device)
 }
 
 
-#define GPIO_DEVICE_INIT(__name, __suffix, __base_addr, __port, __cenr, __bus) \
-static const struct gpio_stm32_config gpio_stm32_cfg_## __suffix = {	\
-	.base = (u32_t *)__base_addr,				\
-	.port = __port,							\
-	.pclken = { .bus = __bus, .enr = __cenr }			\
-};									\
-static struct gpio_stm32_data gpio_stm32_data_## __suffix;		\
-DEVICE_AND_API_INIT(gpio_stm32_## __suffix,				\
-		    __name,						\
-		    gpio_stm32_init,					\
-		    &gpio_stm32_data_## __suffix,			\
-		    &gpio_stm32_cfg_## __suffix,			\
-		    POST_KERNEL,					\
-		    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,			\
+#define GPIO_DEVICE_INIT(__name, __suffix, __base_addr, __port, __cenr, __bus)		\
+static const struct gpio_stm32_config gpio_stm32_cfg_## __suffix = {			\
+	.base = (u32_t *)__base_addr,							\
+	.port = __port,									\
+	.pclken = { .bus = __bus, .enr = __cenr }					\
+};											\
+static struct gpio_stm32_data gpio_stm32_data_## __suffix;				\
+DEVICE_AND_API_INIT(gpio_stm32_## __suffix,						\
+		    __name,								\
+		    gpio_stm32_init,							\
+		    &gpio_stm32_data_## __suffix,					\
+		    &gpio_stm32_cfg_## __suffix,					\
+		    POST_KERNEL,							\
+		    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,				\
 		    &gpio_stm32_driver);
 
 
 #ifdef CONFIG_SOC_SERIES_STM32F1X
 	/* On STM32F1 series, AFIO should be clocked to access GPIOs */
-#define GPIO_DEVICE_INIT_STM32(__suffix, __SUFFIX)			\
-	GPIO_DEVICE_INIT("GPIO" #__SUFFIX, __suffix,			\
-			 GPIO##__SUFFIX##_BASE, STM32_PORT##__SUFFIX,	\
-			 LL_APB2_GRP1_PERIPH_AFIO |			\
-			 STM32_PERIPH_GPIO##__SUFFIX,			\
-			 STM32_CLOCK_BUS_GPIO)
+#define GPIO_DEVICE_INIT_STM32(__suffix, __SUFFIX)				\
+	GPIO_DEVICE_INIT(CONFIG_GPIO_STM32_GPIO##__SUFFIX##_LABEL,		\
+			__suffix,						\
+			CONFIG_GPIO_STM32_GPIO##__SUFFIX##_BASE_ADDRESS, 	\
+			STM32_PORT##__SUFFIX,					\
+			LL_APB2_GRP1_PERIPH_AFIO |				\
```

Comment:
Maybe we can merge both declaration and insert the #ifdef CONFIG_SOC_SERIES_STM32F1X
on this line if this is the only difference.
I think this would be acceptable for ease of read.

Reply:
Seems like a good idea to me.

Reply:
While this change sounds appealing it is not allowed to put #if inside the #define. However the comment could be dropped and the #defines could be merged if I were to do this:
`
#ifdef CONFIG_SOC_SERIES_STM32F1X
#define GPIO_STM32F1X_EXTRA_RCC_BITS  LL_APB2_GRP1_PERIPH_AFIO
#else
#define GPIO_STM32F1X_EXTRA_RCC_BITS  0
#endif /* CONFIG_SOC_SERIES_STM32F1X */
`
In my opinion the code would become self explanatory.